File: Array_Functions/Prefix_Array/Prefix.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> prefixSum(vector<int>& arr, int n) {
    vector<int> res(n);
    res[0] = arr[0];
    for(int i=1;i<n;i++) {
        res[i] = res[i-1] + arr[i];
    }
    return res;
}

int main() {
    int n = 5;
    vector<int> arr = {1, 2, 3, 4, 5};
    
    vector<int> preSum = prefixSum(arr, n);

    for(int i=0;i<n;i++) {
        cout<<preSum[i]<<" ";
    }
    cout<<endl;
}

File: Array_Functions/Prefix_Array/Prefix.py
def prefixSum(arr, n):
    res = [0] * n
    res[0] = arr[0]
    for i in range(1, n):
        res[i] = res[i - 1] + arr[i]
    return res

n = 5
arr = [1, 2, 3, 4, 5]
preSum = prefixSum(arr, n)
for i in range(n):
    print(preSum[i], end = " ")